light-client/src/components/io.rs

```diff
@@ -172,10 +175,11 @@ mod prod {
             }
         }
 
+        // TODO(thane): Generalize over client transport (instead of using HttpClient directly).
         #[pre(self.peer_map.contains_key(&peer))]
-        fn rpc_client_for(&self, peer: PeerId) -> rpc::Client {
+        fn rpc_client_for(&self, peer: PeerId) -> Result<rpc::HttpClient, IoError> {
             let peer_addr = self.peer_map.get(&peer).unwrap().to_owned();
-            rpc::Client::new(peer_addr)
+            Ok(rpc::HttpClient::new(peer_addr).map_err(IoError::from)?)
```

Comment:
Technically a Tendermint `Address` could contain either a TCP or a Unix socket address, and we need the host/port details in order to connect to the RPC endpoint.

We could do away with the need for an error here by requiring host/port details instead of a `peer_addr`, or we could just panic if a user supplies a Unix socket address?